tools/setup-windows/SetupWindows.cs

```diff
@@ -40,31 +40,35 @@ public static int Main (string [] args)
 			var newTargets      = Path.Combine (prefix, "lib", "xamarin.android", "xbuild", "Xamarin", "Android");
 			var refAssemblies   = Path.Combine (progFiles, "Reference Assemblies", "Microsoft", "Framework", "MonoAndroid");
 			var newAssemblies   = Path.Combine (prefix, "lib", "xamarin.android", "xbuild-frameworks", "MonoAndroid");
+			var refAssembliesForVS = Path.Combine (vsInstall, "Common7", "IDE", "ReferenceAssemblies", "Microsoft", "Framework", "MonoAndroid");
 
 			if (Path.DirectorySeparatorChar != '\\') {
 				Console.Error.WriteLine ($"{AppName}: This program is for use on Windows.");
 				return 1;
 			}
 
 			if (args.Length == 0 || args.Any (v => string.Equals (v, "install", StringComparison.OrdinalIgnoreCase) || string.Equals (v, "/install", StringComparison.OrdinalIgnoreCase))) {
-				return Install (msbuildTargets, newTargets, refAssemblies, newAssemblies, hash);
+				return Install (hash, msbuildTargets, newTargets, refAssemblies, newAssemblies, refAssembliesForVS);
 			}
 			if (args.Any (v => string.Equals (v, "uninstall", StringComparison.OrdinalIgnoreCase) || string.Equals (v, "/uninstall", StringComparison.OrdinalIgnoreCase))) {
-				return Uninstall (msbuildTargets, refAssemblies, hash);
+				return Uninstall (hash, msbuildTargets, refAssemblies, refAssembliesForVS);
 			}
 			Console.Error.WriteLine ($"{AppName}: Invalid command `{string.Join (" ", args)}`.");
 			return 1;
 		}
 
-		static int Install (string msbuildTargets, string newTargets, string refAssemblies, string newAssemblies, string hash)
+		static int Install (string hash, string msbuildTargets, string newTargets, string refAssemblies, string newAssemblies, string refAssembliesForVS)
 		{
-			var backupAssemblies    = GetNewBackupName (refAssemblies, hash);
-			var backupTargets       = GetNewBackupName (msbuildTargets, hash);
+			var backupAssemblies        = GetNewBackupName (refAssemblies, hash);
+			var backupAssembliesForVS   = GetNewBackupName (refAssembliesForVS, hash);
+			var backupTargets           = GetNewBackupName (msbuildTargets, hash);
 			try {
 				Directory.CreateDirectory (Path.GetDirectoryName (refAssemblies));
+				Directory.CreateDirectory (Path.GetDirectoryName (refAssembliesForVS));
 				Directory.CreateDirectory (Path.GetDirectoryName (msbuildTargets));
 
 				if (CreateSymbolicLink (refAssemblies, newAssemblies, backupAssemblies) &&
+						CreateSymbolicLink (refAssembliesForVS, newAssemblies, backupAssembliesForVS) &&
```

Comment:
...in particular, I believe that this will *fail* if `%VSINSTALLDIR%` isn't set, as `%ProgramFiles(x86)%\Common7\IDE\ReferenceAssemblies\Microsoft\Framework` *certainly* doesn't exist, and iirc `CreateSymbolicLink()` requires that the directory exist.